be/src/exec/pipeline/exchange/sink_buffer.cpp

```diff
@@ -370,20 +371,45 @@ Status SinkBuffer::_try_to_send_rpc(const TUniqueId& instance_id, const std::fun
 
         closure->cntl.Reset();
         closure->cntl.set_timeout_ms(_brpc_timeout_ms);
-        closure->cntl.request_attachment().append(request.attachment);
 
-        if (bthread_self()) {
-            request.brpc_stub->transmit_chunk(&closure->cntl, request.params.get(), &closure->result, closure);
-        } else {
+        if (!bthread_self()) {
             // When the driver worker thread sends request and creates the protobuf request,
             // also use process_mem_tracker to record the memory of the protobuf request.
             SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(nullptr);
```

Comment:
`SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER` is only effective in the block of `if (!bthread_self()) {...}`, so `_send_rpc` **MUST** be invoked in this block `if (!bthread_self())`.

Reply:
fixed.